src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/AnotherUserProfileActivity.java

```diff
@@ -0,0 +1,235 @@
+
+                            if (valueOfCurrentUserInAnotherUserSocialMap == 0 || valueOfCurrentUserInAnotherUserSocialMap == 2) {
+                                // The logged in user has sent a follow request to this user
+                                changeDisplayToRequestedState();
+                            } else if (valueOfCurrentUserInAnotherUserSocialMap == 1) {
+                                // User on screen is following the logged in user but the logged in
+                                // user hasn't sent the user on screen a follow request
+
+                                followBtn.setOnClickListener(new View.OnClickListener() {
+                                    public void onClick(View v) {
+                                        requestToFollow(currentUserId, anotherUserID, 2);
+                                    }
+                                });
+                            }
+                        } else {
+                            // Case: The logged in user is not following the user on screen and has not sent the
+                            // user on screen a follow request. The user on screen is not following the logged in
+                            // user as well.
+                            followBtn.setOnClickListener(new View.OnClickListener() {
+                                public void onClick(View v) {
+                                    requestToFollow(currentUserId, anotherUserID, 0);
+                                }
+                            });
+                        }
+                    });
+                }
+            } else {
+                // Case: The logged in user is not currently following the user on screen and the
+                // user on screen hasn't sent a follow request to the logged in user
+                // Check if the logged in user has sent a follow request to the user on screen.
+                userController.readUser(anotherUserID, retrievedAnotherUser -> {
+                    if (retrievedAnotherUser.getSocial().containsKey(currentUserId)) {
+                        // Retrieve the value of the logged in user in another user's Social map
+                        int valueOfCurrentUserInAnotherUserSocialMap = ((Number) retrievedAnotherUser.getSocial().get(currentUserId)).intValue();
+
+                        if (valueOfCurrentUserInAnotherUserSocialMap == 0 || valueOfCurrentUserInAnotherUserSocialMap == 2) {
+                            // Case: The logged in user has sent a follow request to this user
+                            changeDisplayToRequestedState();
+                        }
+                    } else {
+                        // Case: The logged in user is not following the user on screen and has not sent the
+                        // user on screen a follow request.
+                        followBtn.setOnClickListener(new View.OnClickListener() {
```

Comment:
You used this followBtn.setOnClickListener code block a few times. Consider moving this to another function where you just pass in (0,1,2). It would help w readability

Reply:
I've refactored the logic based on the (0,1,2) cases that you suggested